src/runtime-rs/crates/resource/src/manager_inner.rs

```diff
@@ -51,6 +54,7 @@ pub(crate) struct ResourceManagerInner {
     pub cgroups_resource: CgroupsResource,
     pub cpu_resource: CpuResource,
     pub mem_resource: MemResource,
+    pub swap_resource: Option<SwapResource>,
```

Comment:
Does swap resource belong to memory resource, which make `swap_resource` as one field of `MemResource` ?

Reply:
The main goal of SwapResource is to dynamically insert swap files into Kata containers on demand, which I believe is somewhat different from MemResource.
I suggest keeping SwapResource and MemResource separate.